analysis/analyzer.py

```diff
@@ -0,0 +1,90 @@
+"""Main binary for the analyis framework.
+
+Takes an input data file and tests various progressive font enrichment methods
+against each one.
+
+Collects an overall score for each method and reports the aggregate results.
+
+Input data is in textproto format using the proto definitions found in
+analysis/page_view_sequence.proto
+"""
+
+from absl import app
+from absl import flags
+from analysis import page_view_sequence_pb2
+from analysis import simulation
+from analysis import fake_pfe_method
+from google.protobuf import text_format
+from patch_subset.py import patch_subset_method
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string("input_data", None, "Path to input data for the analysis.")
+flags.mark_flag_as_required("input_data")
+
+flags.DEFINE_string(
+    "font_directory", None,
+    "Directory which contains all font's to be used in the analysis.")
+flags.mark_flag_as_required("font_directory")
+
+PFE_METHODS = [
+    fake_pfe_method,
+    patch_subset_method,
+]
+
+NETWORK_MODELS = [
+    # TODO(garretrieger): populate with some real network models.
+    simulation.NetworkModel(
+        "broadband",
+        rtt=20,  # 40 ms
+        bandwidth_up=1250,  # 10 mbps
+        bandwidth_down=6250),  # 50 mbps
+    simulation.NetworkModel(
+        "dialup",
+        rtt=200,  # 200 ms
+        bandwidth_up=7,  # 56 kbps
+        bandwidth_down=7),  # 56 kbps
+]
+
+
+def cost(time_ms):
+  """Assigns a cost to a measure of request latency."""
+  # TODO(garretrieger): implement me.
+  return time_ms
+
+
+def analyze_data_set(data_set, pfe_methods, network_models, font_directory):
```

Comment:
enough parts a single type (maybe just a namedtuple) capturing all the input bits might make sense?

Reply:
Not sure adding a single type helps, since this function is typically going be called only once per program invocation.